sources/models/src/modeled_types/credentialproviders.rs

```diff
@@ -0,0 +1,154 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+// Just need serde's Error in scope to get its trait methods
+use super::error;
+use serde::de::Error as _;
+use snafu::ensure;
+use std::borrow::Borrow;
+use std::convert::TryFrom;
+use std::fmt;
+use std::ops::Deref;
+
+use crate::{SingleLineString, ValidBase64};
+
+/// CacheDurationValue is a string that represents the duration the plugin will
+/// cache credentials in-memory if a cache duration is not provided in the
+/// plugin response.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CacheDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    // A duration string is a sequence of decimal numbers, each with a
+    // unit suffix, such as "300ms" or "2h45m".
+    // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+    // Based on restraints from Go's `ParseDuration` funciton.
+    pub(crate) static ref DURATION_FORMAT: Regex = Regex::new(r"^([0-9]*[a-z]+)+$").unwrap();
+}
+
+impl TryFrom<&str> for CacheDurationValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        ensure!(
+            DURATION_FORMAT.is_match(input),
+            error::InvalidCacheDurationValueSnafu { input }
+        );
+        Ok(CacheDurationValue {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(CacheDurationValue, "CacheDurationValue");
+
+#[cfg(test)]
+mod test_cacheduration_value {
+    use super::CacheDurationValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_cacheduration_value() {
+        for ok in &[
+            "12h",
+            "120m",
+            "2h15m"
+        ] {
+            CacheDurationValue::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn invalid_cacheduration_value() {
+        for err in &[
+            "12 days",
+            "💝",
+            "Ï",
+            "",
+        ] {
+            CacheDurationValue::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// CredentialPluginName is the name of the plugin which MUST match the
+/// name of the executable binary that exists in the Kubelet
+/// `--image-credential-provider-bin-dir` directory. 
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CredentialPluginName {
+    inner: String,
+}
+
+lazy_static! {
+    // List of valid, supported credential provider plugins. Update this list
+    // if/when we ever add or remove credential provider plugin binaries.
+    pub(crate) static ref CREDENTIALPROVIDER_PLUGINS: Vec<&'static str> = vec!["ecr-credential-provider"];
+}
```

Comment:
Doesn't this work for you?

```suggestion
pub(crate) const CREDENTIALPROVIDER_PLUGINS: [&str; 1] = ["ecr-credential-provider"];
```

Reply:
Great, that works. Still learning. ;)

Reply:
Nice! I'm learning too (I wasn't sure this would fit your use case)

Reply:
One small change I did need to make. Instead of:

```
[&str; 1]
```

I needed:

```
[&'static str; 1]
```